schema.ts

```diff
@@ -200,33 +213,74 @@ export type NumberSchema =
   | NumericDateTimeSchema;
 
 export interface BaseNumberSchema<T extends NumberHintTypes = NumberHintTypes> extends BaseSchema {
+  /** Identifies this schema as relating to a number value. */
   type: ValueType.Number;
+  /** An optional type hint instructing Coda about how to interpret or render this value. */
   codaType?: T;
 }
 
+/**
+ * A schema representing a return value or object property that is a numeric value,
+ * i.e. a raw number with an optional decimal precision.
+ */
 export interface NumericSchema extends BaseNumberSchema {
+  /** If specified, instructs Coda to render this value as a percentage. */
   codaType?: ValueHintType.Percent; // Can also be undefined if it's a vanilla number
+  /** The decimal precision. The number will be rounded to this precision when rendered. */
   precision?: number;
+  /** If specified, will render thousands separators for large numbers, e.g. `1,234,567.89`. */
   useThousandsSeparator?: boolean;
 }
 
+/**
+ * A schema representing a return value or object property that is provided as a number,
+ * which Coda should interpret as a date. The given number should be in seconds since the Unix epoch.
+ */
 export interface NumericDateSchema extends BaseNumberSchema<ValueHintType.Date> {
+  /** Instructs Coda to render this value as a date. */
   codaType: ValueHintType.Date;
-  // A Moment date format string, such as 'MMM D, YYYY', that corresponds to a supported Coda date column format.
+  /**
+   * A Moment date format string, such as 'MMM D, YYYY', that corresponds to a supported Coda date column format.
+   *
+   * Only applies when this is used as a sync table property.
+   */
   format?: string;
```

Comment:
This API is currently kind of crappy. The main app represents format options for date/time/datetime columns as a preset list Moment format strings, like 'MMM D, YYYY'. In practice, this is an enum, since there are a specific set of valid values.

The reason this was added was for Cross-doc, which gets the format string of the source table via the API, and then passes it through as the schema of the destination table. The API doesn't attempt to enumerate all the formats, it just knows that they're strings. And theoretically DocX folks could add new format strings at any time. So we were trying to eliminate the need to update the API or SDK when that happens.

But for now it means that unless you use one of the magic format strings that the app supports, your formatting will be ignored. (And it will already be ignored if this isn't a sync table property, since canvas values don't support formatting yet.)

I don't know if we want to aim to convert this to an enum, or explain this, or list the valid values somewhere, or something else.

Reply:
I agree that an enum is problematic. The Sheets API also had a string format, and we just documented that format via handwritten docs:

https://developers.google.com/sheets/api/guides/formats

Once we have that we can come back and update these reference docs to point to that page.

Reply:
Aha great, glad there's precedence for this.